spec/spec.md

```diff
@@ -511,6 +511,180 @@ sequenceDiagram
+
+::: todo Request Presentation
+See [#21](https://github.com/decentralized-identity/waci-presentation-exchange/issues/21)
+:::
+
+```
+{
+  "type": "https://didcomm.org/present-proof/3.0/request-presentation",
+  "id": "0ac534c8-98ed-4fe3-8a41-3600775e1e92",
+  "from": "did:example:verifier",
+  "to": "did:example:prover",
+  "body": {},
+  "attachments": [
+    {
+      "@id": "ed7d9b1f-9eed-4bde-b81c-3aa7485cf947",
+      "mime-type": "application/json",
+      "format": "dif/presentation-exchange/definitions@v1.0",
+      "data": {
+        "json": {
+          "dif": {
+            "options": {
+              "challenge": "3fa85f64-5717-4562-b3fc-2c963f66afa7",
+              "domain": "4jt78h47fh47"
+            },
+            "presentation_definition": {
+              "frame": {
+                  "@context": [
+                    "https://www.w3.org/2018/credentials/v1",
+                    "https://w3id.org/vaccination/v1",
+                    "https://w3id.org/security/suites/bls12381-2020/v1"
+                  ],
+                  "type": ["VerifiableCredential", "VaccinationCertificate"],
+                  "credentialSubject": {
+                    "@explicit": true,
+                    "type": ["VaccinationEvent"],
+                    "batchNumber": {},
+                    "countryOfVaccination": {}
+                  }
+                },
+                "input_descriptors": [
+                  {
+                    "id": "vaccination_input",
+                    "name": "Vaccinatin Certificate",
+                    "schema": "https://w3id.org/vaccination/#VaccinationCertificate",
+                    "constraints": {
+                      "fields": [
+                        {
+                          "path": ["$.credentialSubject.batchNumber"],
+                          "filter": {
+                            "type": "string"
+                          }
+                        },
+                        {
+                          "path": ["$.credentialSubject.countryOfVaccination"],
+                          "filter": {
+                            "type": "string"
+                          }
+                        }
+                      ]
+                    }
+                  }
+                ]
+            }
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+### Message 2 - Present Proof
+
+::: todo Present Proof
+See [#22](https://github.com/decentralized-identity/waci-presentation-exchange/issues/22)
+:::
+
+```
+{
+  "type": "https://didcomm.org/present-proof/3.0/presentation",
+  "id": "f1ca8245-ab2d-4d9c-8d7d-94bf310314ef",
+  "from": "did:example:verifier",
+  "to": "did:example:prover",
+  "body": {},
+  "attachments": [
+    {
+      "@id": "2a3f1c4c-623c-44e6-b159-179048c51260",
+      "mime-type": "application/ld+json",
+      "format": "dif/presentation-exchange/submission@v1.0",
+      "data": {
+        "@context": [
+          "https://www.w3.org/2018/credentials/v1"
```

Comment:
Does this need the Presentation Submission context and type entries?

Reply:
excellent catch, you are correct.